examples/motoko_examples/threshold_ecdsa/canisters/rust/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use ic_cdk::export::{
+    candid::CandidType,
+    serde::{Deserialize, Serialize},
+    Principal,
+};
+use ic_cdk_macros::*;
+use std::cell::RefCell;
+use std::str::FromStr;
+
+//#region Performance
+#[derive(candid::CandidType, Clone)]
+pub struct PerfResult {
+    wasm_body_only: u64,
+    wasm_including_prelude: u64
+}
+
+thread_local! {
+    static PERF_RESULT_REF_CELL: RefCell<Option<PerfResult>> = RefCell::default();
+}
+
+#[ic_cdk_macros::query]
+pub fn get_perf_result() -> Option<PerfResult> {
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| perf_result_ref_cell.borrow().as_ref().cloned())
+}
+
+fn record_performance(start: u64, end: u64) {
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| {
+        let mut perf_result = perf_result_ref_cell.borrow_mut();
+
+        *perf_result = Some(PerfResult {
+            wasm_body_only: end - start,
+            wasm_including_prelude: ic_cdk::api::call::performance_counter(0)
+        })
+    });
+}
+//#endregion
+
+#[derive(CandidType, Serialize, Debug)]
+struct PublicKeyReply {
+    pub public_key: Vec<u8>,
+}
+
+#[derive(CandidType, Serialize, Debug)]
+struct SignatureReply {
+    pub signature: Vec<u8>,
+}
+
+type CanisterId = Principal;
+
+#[derive(CandidType, Serialize, Debug)]
+struct ECDSAPublicKey {
+    pub canister_id: Option<CanisterId>,
+    pub derivation_path: Vec<Vec<u8>>,
+    pub key_id: EcdsaKeyId,
+}
+
+#[derive(CandidType, Deserialize, Debug)]
+struct ECDSAPublicKeyReply {
+    pub public_key: Vec<u8>,
+    pub chain_code: Vec<u8>,
+}
+
+#[derive(CandidType, Serialize, Debug)]
+struct SignWithECDSA {
+    pub message_hash: Vec<u8>,
+    pub derivation_path: Vec<Vec<u8>>,
+    pub key_id: EcdsaKeyId,
+}
+
+#[derive(CandidType, Deserialize, Debug)]
+struct SignWithECDSAReply {
```

Comment:
Use consistent naming of Ecdsa

Reply:
I suppose you just copied this example over, so if that's the case then we'll just keep it as it is